### PR TITLE
Use canonical URL for Maven Central repository shortcut method

### PR DESCRIPTION
**Pull request by jvasallo**
### Context
The default Maven URL is https://repo.maven.apache.org/maven2/. (https://repo1.maven.org/maven2/ is no longer the default)

From reading online, As Robert Scholte informed, as of 3.0.4 the default URL has been changed from http://repo1.maven.org/maven2 to http://repo.maven.apache.org/maven2. 

Additional Sources:
- http://svn.apache.org/viewvc/maven/site/trunk/src/site/xdoc/repository/index.xml?r1=1243210&r2=1243209&pathrev=1243210
- https://stackoverflow.com/questions/18471432/maven-why-does-my-computer-always-download-from-http-repo1-maven-org
- https://github.com/gradle/gradle/issues/3463

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/.github/CONTRIBUTING.md)
- [x] [Sign Gradle CLA](http://gradle.org/contributor-license-agreement/)
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by bmuschko** [comment]
Did you already submit a signed CLA? I couldn't find it. What name did you use?

**Comment by jvasallo** [comment]
@bmuschko I signed it about 20-30 minutes ago as Joel Vasallo; maybe its not updated? I did it both as an individual and as a contributor with a corporation. This is my personal account at the moment; signed both just in case.

**Comment by bmuschko** [comment]
@jvasallo Thanks, just got it.

**Comment by bmuschko** [comment]
There are also test or test-related classes that should adapt the new URL:

- `ApacheDirectoryListingParserTest."parse is compatible with #repoType"`
- `MavenInstallationDownloader.downloadMavenBinArchiveWithRetry(String)`.

File: subprojects/resources-http/src/test/groovy/org/gradle/internal/resource/transport/http/ApacheDirectoryListingParserTest.groovy [hunk 1]
```diff
@@ -165,7 +165,7 @@ class ApacheDirectoryListingParserTest extends Specification {
         where:
         artifactRootURI                                                         | repoType
         "http://localhost:8081/artifactory/repo1/junit/junit/"                  | "artifactory"
-        "https://repo1.maven.org/maven2/junit/junit/"                           | "mavencentral"
+        "https://repo.maven.apache.org/maven2/junit/junit/"                           | "mavencentral"
```
Comment (bmuschko) [inline subprojects/resources-http/src/test/groovy/org/gradle/internal/resource/transport/http/ApacheDirectoryListingParserTest.groovy]: Can you align the pipe character with the other ones?

Comment (jvasallo) [reply subprojects/resources-http/src/test/groovy/org/gradle/internal/resource/transport/http/ApacheDirectoryListingParserTest.groovy]: Fixed. Sorry missed that.

**Comment by jvasallo** [comment]
Fixed the existing test cases/documentation I could find with references to `repo1`. I ran a local `./gradlew coreApi:check` test and it seems to be in order. 

Anything else you would like me to do?

**Comment by bmuschko** [comment]
@jvasallo Thanks, I think there's one more place in `depMngmt.adoc` that references a link to `https://repo1.maven.org/maven2`. Could you touch that up as well?

**Comment by jvasallo** [comment]
Blah! Missed one. Sorry about that; fixed it now.

I did a recursive search and couldn't find any other references but let me know if there is anything else! 👍

**Comment by bmuschko** [comment]
Thanks for providing the pull request. It has been merged.

**Pull request merged**